### DISCUSSION TITLE
When the size of the ringbuffer is not an integer multiple of 4096, the syscall return EINVAL.

### DISCUSSION THREAD

**Discussion by hengyoush** [cilium/ebpf#1407] [Q&A] [answered]
libbpf avoids this issue by adjusting the size to an integer multiple of 4096 before the syscall, so does ebpfgo need to perform the same handling?

**Comment by lmb** [comment]
The value of 4096 is probably not a constant, I'm guessing it's the size of an operating system page? I'm always very careful about enforcing another system's (here the kernel) constraints, since it can cause problems if the kernel ever decides to change the limit. To sum up, I think that silently changing the value is a bad idea.

We could / should however make this easier to figure out for users. A rough idea: if the syscall returns `EINVAL` and the size is not a multiple of the page size (this needs digging into the kernel to confirm) amend the error to be something like:

```
perf reader: EINVAL (size is not a multiple of page size?)
```

**Reply by hengyoush** [reply]
Thank you for your reply. After further reviewing the libbpf source code, I discovered that the value of 4096 is indeed the page size obtained from `sysconf(_SC_PAGE_SIZE)`.
The real problem is that the error logs do not provide useful information😂. As you mentioned, we could provide more user-friendly prompts to help users. If possible, I would like to submit a pull request (PR) .

**Reply by lmb** [reply]
Please do!

**Comment by florianl** [comment, marked as answer]
Issue got resolved with the merge of https://github.com/cilium/ebpf/pull/1420.
